### PR TITLE
chore(pie-monorepo): DSW-1022 add js linting to components

### PR DESCRIPTION
**Pull request by jamieomaguire**
---
"@justeattakeaway/pie-webc-testing": minor
"@justeattakeaway/pie-icon-button": minor
"@justeattakeaway/pie-webc-core": minor
"@justeattakeaway/pie-wrapper-react": minor
"@justeattakeaway/pie-button": minor
"@justeattakeaway/pie-modal": minor
"pie-monorepo": minor
---

[Changed] - Add js/ts linting and fix errors

### CODE REVIEW

File: packages/components/pie-button/.eslintignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.turbo
```
Comment (jamieomaguire) [inline packages/components/pie-button/.eslintignore:1]: A note here - I tried adding a single `.eslintignore` to the root of the repo but it didn't seem to get picked up/work

File: packages/components/pie-button/package.json [hunk 2]
```diff
@@ -8,6 +8,8 @@
   "types": "dist/types/index.d.ts",
   "scripts": {
     "build": "yarn build:wrapper pie-button && run -T vite build ",
+    "lint:scripts": "run -T eslint .",
+    "lint:scripts:fix": "run -T eslint . --fix",
```
Comment (raoufswe) [inline packages/components/pie-button/package.json:12]: can you also please add these to the component generator if the update is across all components?

File: packages/components/pie-button/playwright-lit-visual.config.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import { defineConfig } from '@sand4rt/experimental-ct-web';
 import { getPlaywrightVisualConfig } from '@justeattakeaway/pie-components-config';
 
-// @ts-ignore
```
Comment (raoufswe) [inline packages/components/pie-button/playwright-lit-visual.config.ts:4]: same here

File: packages/components/pie-button/package.json (hunk 2 above)
Comment (jamieomaguire) [reply packages/components/pie-button/package.json:12]: I'll do that in a separate PR if you don't mind - I'll raise it after this one :)

**Comment by raoufswe** [comment]
good stuff!
could you please ensure these changes are also reflected in the generator? 🥇

**Review by raoufswe — APPROVED** [review]

**Review by LTurns — APPROVED** [review]

File: packages/tools/pie-wrapper-react/custom-elements.json [hunk 4]
```diff
@@ -442,7 +442,7 @@
                                     "text": "TemplateResult | typeof nothing"
                                 }
                             },
-                            "description": "Render supportingAction button depending on prop availability.\n\n1. If the prop `supportingAction` is not provided, the button is not rendered.\n2. If the prop `supportingAction` is provided but any of the optional properties\nare not provided, they fall back to their default values.\n\n3. If `supportingAction` is provided but not `leadingAction`, log a warning."
+                            "description": "Render supportingAction button depending on prop availability.\n\n1. If the prop `supportingAction` is not provided, the button is not rendered.\n2. If the prop `supportingAction` is provided but any of the optional properties\nare not provided, they fall back to their default values.\n3. If `supportingAction` is provided but not `leadingAction`, log a warning and do\nnot render `supportingAction`."
```
[the remark this answers is not in the document]
Comment (jamieomaguire) [reply packages/tools/pie-wrapper-react/custom-elements.json:445]: This isn't something I have control over so we should open a new ticket to look into this

[the remark this answers is not in the document]
Comment (raoufswe) [reply packages/tools/pie-wrapper-react/custom-elements.json:445]: never mind, just a slow morning. `\n` is an escape character, and i thought it's don't render 😮‍💨

File: packages/components/pie-button/package.json (hunk 2 above)
Comment (ashleynolan) [reply packages/components/pie-button/package.json:12]: Quick note – this could just be `yarn lint:scripts --fix` (for that follow up PR)

**Pull request merged**